packages/x-components/src/plugins/x-emitters.ts

```diff
@@ -57,14 +58,15 @@ export function registerStoreEmitters(
  */
 function normalizeStateSelector(
   stateSelector: AnySimpleStateSelector | AnyStateSelector
-): Required<AnyStateSelector> {
+): Required<Omit<AnyStateSelector, keyof WatchOptionsBase>> {
```

Comment:
StateSelector extends WatchOptions which in Vue3 adds once prop and also extends WatchOptionsBase. To overcome this type error, I'm omitting the WatchOptionsBase props from being required.

Reply:
I am not sure if even removing the returning type, we would do less type-overengineering... 
Perhaps making also a constant with the default options could help to understand.
```TS
const defaultOptions: Omit<AnyStateSelector, 'selector'> = {
  deep: false,
  immediate: false,
  once: false,
  filter: () => true,
  metadata: { replaceable: true }
};
```
I haven't got a strong opinion here. Whatever you decide, I approve ✌🏾

Reply:
I also have mixed feelings about this. But I'm happy with my proposal because in the end is simple. Don't know if  @diegopf has a different opinion about this.

Reply:
Both approaches seem fine to me. I kind of prefer Jose's approach as it appears easier.